### PR TITLE
Graphql: Add flights query

### PR DESCRIPTION
**Pull request by tbergq**

### CODE REVIEW

File: packages/graphql/index.js [hunk 1]
```diff
@@ -9,20 +9,21 @@ import {
...
 
 import fetch from './src/services/Fetch';
+import AllFlights from './src/queries/AllFlights';
+import createContext from './src/services/GraphQLContext';
 
 const schema = new GraphQLSchema({
   query: new GraphQLObjectType({
     name: 'RootQueryType',
     fields: {
+      allFlights: AllFlights,
```
Comment (michalsanger) [inline packages/graphql/index.js]: I hope it's a right time to discuss some naming conventions. I suggest not to use prefix `all`, we stoped using it on graphql.kiwi.com it became something like "habit you can't explain"

File: packages/graphql/src/queries/AllFlights.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  +input: {|
+    +flyFrom: string,
+    +dateFrom: Date,
+    +dateTo?: Date,
+  |},
+  ...$Exact<ConnectionArguments>,
+|};
+
+export default {
+  name: 'AllFlights',
```
Comment (michalsanger) [inline packages/graphql/src/queries/AllFlights.js]: We should not forget we are not about flights anymore. This query can return also trains and buses, in the future also ferries etc.

Based on [our gossary](https://kiwi.wiki/kiwi/handbook/#/comms/glossary?id=itinerary-parts-examples) this query returns itineraries so shall we named it `itineraries` or `searchItineraries` maybe?

File: packages/graphql/index.js (hunk 1 above)
Comment (FilipMessa) [reply packages/graphql/index.js]: I agree with you prefix **all** is kind of confusing. And based naming conventions on kiwi's glossary make sense. I would like to share this naming convention as a part of our documentation.

Comment (tbergq) [reply packages/graphql/index.js]: I agree, I will come up with a better name

File: packages/graphql/src/queries/AllFlights.js (hunk 2 above)
Comment (tbergq) [reply packages/graphql/src/queries/AllFlights.js]: I think itineraries is fine, I will change it

**Comment by tbergq** [comment]
@FilipMessa @michalsanger 🚀

File: packages/graphql/src/queries/Itineraries.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    },
+    ...connectionArgs,
+  },
+  type: ItinerariesConnection,
+  resolve: async (_: mixed, args: Args, { dataLoader }: GraphqlContextType) => {
+    const {
+      input: { flyFrom, dateFrom, dateTo },
+    } = args;
+    const flights = await dataLoader.itineraries.load({
+      flyFrom,
```
[the remark this answers is not in the document]
Comment (tbergq) [reply packages/graphql/src/queries/Itineraries.js]: Yeah, this is actually what the parameter is called when we pass it to tequila, but sure, we could change it too

**Comment by tbergq** [comment]
@FilipMessa 🚀

**Comment by FilipMessa** [comment]
There is still some naming with `flights` so because we assume that in futter it won't be only about flights, we should change all these names.
What do you think?

**Comment by tbergq** [comment]
🙈 you are right, I missed this, I will change it.

**Comment by tbergq** [comment]
@FilipMessa should be no flights left now

**Comment by FilipMessa** [comment]
Thanks, good job.
But it hard to know if it works without Playground or some FE implementation now, so what about adding there some tests at least?

**Comment by tbergq** [comment]
Yes, I wanted to add some tests, but we need some Testing tool, which I think should be another PR, this is big enough already. 

I have an implementation I used to test it which I did not commit. Do you want me to add that?

**Comment by FilipMessa** [comment]
OK, sure make tests in another PR than, I believe you that it works so. I'll approve this PR.

File: scripts/createSchema.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+import { schema } from '@kiwicom/margarita-graphql';
+import { printSchema } from 'graphql';
+import fs from 'fs';
+import path from 'path';
+
+fs.writeFileSync(
+  path.join(__dirname, '..', 'schema.graphql'),
+  printSchema(schema),
+);
```
Comment (mrtnzlml) [inline scripts/createSchema.js]: 🤔 https://github.com/kiwicom/margarita/blob/06455a179940dc4688029d05a131701eb65bf504/scripts/relay-compiler.js#L20

The reason why I added it together with Relay Compiler is that it always fetches (reads) new schema before Relay works with it so you don't even have to think about such a step. This script should not be necessary.

Comment (tbergq) [reply scripts/createSchema.js]: Yes, I was unaware of this when making the PR. I will remove it.

**Pull request merged**